app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -266,10 +266,21 @@ public IdentifierHolder signUp(Study study, StudyParticipant participant) {
         } catch(EntityAlreadyExistsException e) {
             // Suppress this and send an email to notify the user that the account already exists. From 
             // this call, we simply return a 200 the same as any other sign up. Otherwise the response 
-            // reveals that the email has been taken.
-            AccountId accountId = AccountId.forId(study.getIdentifier(), (String) e.getEntityKeys().get("userId"));
-            accountWorkflowService.notifyAccountExists(study, accountId);
-            LOG.info("Sign up attempt for existing email address in study '"+study.getIdentifier()+"'");
+            // reveals that the credential has been taken.
+            AccountId accountId = null;
+            if (e.getEntityKeys().get("email") != null) {
+                accountId = AccountId.forEmail(study.getIdentifier(), participant.getEmail());
+            } else if (e.getEntityKeys().get("phone") != null) {
+                accountId = AccountId.forPhone(study.getIdentifier(), participant.getPhone());
+            } else if (e.getEntityKeys().get("externalId") != null) {
+                accountId = AccountId.forExternalId(study.getIdentifier(), participant.getExternalId());
+            }
```

Comment:
Add unit tests for each of the 3 cases, as well as the case where no account is found.